### PR TITLE
AJ-586: feature flag for WDS

### PR DESCRIPTION
**Pull request by davidangb**
Scope of this PR is to:
1. Create a feature flag, available at #feature-preview, to control WDS data tables
2. Show a visual indication if the feature flag is enabled

Out of scope for this PR is any actual connection to WDS. We're not displaying any real WDS data; this PR is simply about the feature flag.

Feature flag:
Screenshot 9-27-2022 at 03 44 PM (1)

with flag disabled (no changes compared to current UI):
Screenshot 9-27-2022 at 03 44 PM (2)

with flag enabled:
Screenshot 9-27-2022 at 03 44 PM

### CODE REVIEW

File: src/libs/feature-previews-config.js [hunk 1]
```diff
@@ -12,6 +12,13 @@ const featurePreviewsConfig = [
     description: 'Enabling this feature will allow you to view information about the workflow that generated data table columns and files.',
     groups: ['preview-data-versioning-and-provenance'],
     feedbackUrl: `mailto:dsp-sue@broadinstitute.org?subject=${encodeURIComponent(`Feedback on data table provenance`)}`
+  },
+  {
+    id: 'workspace-data-service',
+    title: 'Workspace Data Service (WDS) on Azure',
+    description: 'Enabling this feature will enable WDS-powered data tables.',
+    groups: ['preview-wds-on-azure'],
+    feedbackUrl: `mailto:dsp-analysis-journeys@broadinstitute.org?subject=${encodeURIComponent(`Feedback on WDS UI`)}`
```
Comment (davidangb) [inline src/libs/feature-previews-config.js:21]: I have also created the `preview-wds-on-azure` managed group in prod. We don't need to create that group in non-prod envs:
https://github.com/DataBiosphere/terra-ui/blob/248d67f74a595aab8f674fab1f09dc5930708451/src/libs/feature-previews.js#L31-L32

File: src/pages/workspaces/workspace/Data.js [hunk 2]
```diff
@@ -700,6 +700,18 @@ const WorkspaceData = _.flow(
...
+            }, [
+              div({
+                style: {
+                  display: 'flex', flexDirection: 'column', alignItems: 'flex-start',
+                  padding: '0.5rem 1.5rem', borderBottom: `1px solid ${colors.dark(0.2)}`,
+                  backgroundColor: 'white'
+                }
+              }, ['Coming soon.']),
+              div({}, '')
+            ]),
```
Comment (davidangb) [inline src/pages/workspaces/workspace/Data.js:714]: `DataTypeSection` only seems to display its children when there are > 1 children, so I added the extra `div({}, '')`. This is throwaway code which will soon be replaced by data coming from WDS, so I don't mind the weirdness here.

**Review by calypsomatic — APPROVED** [review]

**Review by cahrens — APPROVED** [review]

File: src/pages/workspaces/workspace/Data.js (hunk 2 above)
Comment (breilly2) [reply src/pages/workspaces/workspace/Data.js:714]: Maybe [this](https://github.com/DataBiosphere/terra-ui/blob/3a84398c526a71a07eae371fe22ba84a2fe56c44/src/pages/workspaces/workspace/Data.js#L225)? Probably `children` should be assumed to already be an array, so it shouldn't need to be in additional brackets (`[children]`).

Comment (nawatts) [reply src/pages/workspaces/workspace/Data.js:714]: > Probably children should be assumed to already be an array

`children` should generally be treated as "an opaque data structure".
https://reactjs.org/docs/react-api.html#reactchildren

**Pull request merged**